pkg/noderesourcetopology/filter.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	v1qos "k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
+	bm "k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "NodeResourceTopologyMatch"
+)
+
+var _ framework.FilterPlugin = &TopologyMatch{}
+
+type nodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type PolicyHandler func(pod *v1.Pod, zoneMap topologyv1alpha1.ZoneList) *framework.Status
+
+type PolicyHandlerMap map[apiconfig.TopologyManagerPolicy]PolicyHandler
+
+// TopologyMatch plugin which run simplified version of TopologyManager's admit handler
+type TopologyMatch struct {
+	policyHandlers PolicyHandlerMap
+	lister         listerv1alpha1.NodeResourceTopologyLister
+	namespace      string
+}
+
+type NUMANode struct {
+	NUMAID    int
+	Resources v1.ResourceList
+}
+
+type NUMANodeList []NUMANode
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (tm *TopologyMatch) Name() string {
+	return Name
+}
+
+// getTopologyPolicies return true if we're working with such policy
+func getTopologyPolicies(nodeTopology *topologyv1alpha1.NodeResourceTopology, nodeName string) []apiconfig.TopologyManagerPolicy {
+	policies := make([]apiconfig.TopologyManagerPolicy, 0)
```

Comment:
It's conventional in Go to init a slice using:

```suggestion
	var policies[]apiconfig.TopologyManagerPolicy
```